ethers-core/src/types/chain.rs

```diff
@@ -60,6 +61,39 @@ pub enum Chain {
     AuroraTestnet = 1313161555,
 }
 
+// === impl Chain ===
+
+impl Chain {
+    /// The blocktime varies from chain to chain
+    ///
+    /// It can be beneficial to know the average blocktime to adjust the polling of an Http provider
+    /// for example.
+    ///
+    /// **Note:** this will not return the accurate average depending on the time but is rather a
+    /// sensible default derived from blocktime charts like <https://etherscan.com/chart/blocktime>
+    /// <https://polygonscan.com/chart/blocktime>
+    pub fn average_blocktime_hint(&self) -> Option<Duration> {
+        let ms = match self {
+            Chain::Arbitrum | Chain::ArbitrumTestnet => 1_300,
+            Chain::Mainnet | Chain::Optimism => 13_000,
+            Chain::Polygon | Chain::PolygonMumbai => 2_100,
+            Chain::Moonbeam | Chain::Moonriver => 12_500,
+            Chain::BinanceSmartChain | Chain::BinanceSmartChainTestnet => 3_000,
+            Chain::Avalanche | Chain::AvalancheFuji => 2_000,
+            Chain::Fantom | Chain::FantomTestnet => 1_200,
+            Chain::Cronos | Chain::CronosTestnet => 5_700,
+            Chain::Evmos | Chain::EvmosTestnet => 1_900,
+            Chain::Aurora | Chain::AuroraTestnet => 1_100,
+            Chain::Oasis => 5_500,
+            Chain::Emerald => 6_000,
+            Chain::Dev | Chain::AnvilHardhat => 200,
+            _ => return None,
```

Comment:
I think it is better to explicitly catch other variants, otherwise it will be very easy for others contributor that are adding new networks to miss this match!

Reply:
What should we do? panic?

Reply:
sorry I wasn't clear. Still return None is fine, but instead of using `_` as a pattern, write down all other enum variants.

This way if a new network is added, compiler will help developer noticing that this is something to fill in.

Reply:
reasonable, do you want to add this?

Reply:
https://github.com/gakonst/ethers-rs/pull/1635